server/src/test/java/org/bonitasoft/console/common/server/servlet/IconServletTest.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2016 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This library is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU Lesser General Public License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+ **/
+
+package org.bonitasoft.console.common.server.servlet;
+
+import static org.mockito.Mockito.*;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.bonitasoft.engine.api.IdentityAPI;
+import org.bonitasoft.engine.exception.NotFoundException;
+import org.bonitasoft.engine.identity.impl.IconImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * @author Baptiste Mesta
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class IconServletTest {
+
+    private static final long ICON_ID = 1238970432L;
+    private static final String ICON_ID_STRING = "1238970432";
+    private static final byte[] ICON_CONTENT = "content".getBytes();
+    @InjectMocks
+    @Spy
+    private IconServlet iconServlet;
+    @Mock
+    private HttpServletRequest httpServletRequest;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Mock
+    private IdentityAPI identityAPI;
+    @Mock
+    private ServletOutputStream outputStream;
+
+    @Before
+    public void before() throws Exception {
+        doReturn(identityAPI).when(iconServlet).getIdentityApi(httpServletRequest);
+        doReturn(outputStream).when(httpServletResponse).getOutputStream();
+    }
+
+    private void havingIcon(String mimeType, long iconId) throws NotFoundException {
+        doReturn(new IconImpl(iconId, mimeType, ICON_CONTENT)).when(identityAPI).getIcon(iconId);
+    }
+
+    private HttpServletRequest request(String contextUrl) {
+        doReturn(contextUrl).when(httpServletRequest).getPathInfo();
+        return httpServletRequest;
+    }
+
+    @Test
+    public void should_get_return_user_icon_from_engine() throws Exception {
```

Comment:
`should_get_return_user_icon_from_engine` this is not an English sentence. 
We want to test object's behavior and the subject of `should` is the object under test. So for you case this is the IconServlet that should do things.
Test names should be as follow
`should do something when something`